docs/Top50ServiceChecker.ipynb

```diff
@@ -69,155 +76,249 @@
+    "    \"MOD021KM\": \"C1378227407-LAADS\",\n",
+    "    \"MOD06_L2\": \"C1443535037-LAADS\",\n",
+    "    \"VNP46A1\": \"C1897815356-LAADS\",\n",
+    "    \"MOD07_L2\": \"C1443541366-LAADS\",\n",
+    "    \"MOD05_L2\": \"C1443531026-LAADS\",\n",
+    "    \"MOD02HKM\": \"C1378577630-LAADS\",\n",
+    "    \"MCD06COSP_D3_MODIS\": \"C1887589686-LAADS\",\n",
+    "    \"MYD04_3K\": \"C1443528505-LAADS\",\n",
+    "    \"MYD07_L2\": \"C1443544285-LAADS\",\n",
+    "    \"MOD04_3K\": \"C1443420430-LAADS\",\n",
+    "    \"MOD35_L2\": \"C1443561895-LAADS\",\n",
+    "    \"MYD05_L2\": \"C1443531357-LAADS\",\n",
+    "    \"MOD11A1\": \"C1748058432-LPCLOUD\",\n",
+    "    \"MYD11A1\": \"C1748046084-LPCLOUD\",\n",
+    "    \"MOD11A2\": \"C2269056084-LPCLOUD\",\n",
+    "    \"MYD11A2\": \"C2269057787-LPCLOUD\",\n",
+    "    \"MCD19A2\": \"C2324689816-LPCLOUD\",\n",
+    "    \"MYD09GA\": \"C2202498116-LPCLOUD\",\n",
+    "    \"MOD15A2H\": \"C2218777082-LPCLOUD\",\n",
+    "    \"MOD09GA\": \"C2202497474-LPCLOUD\",\n",
+    "    \"MCD43A4\": \"C2218719731-LPCLOUD\",\n",
+    "    \"MOD14\": \"C2271754179-LPCLOUD\",\n",
+    "    \"MCD15A2H\": \"C2222147000-LPCLOUD\",\n",
+    "    \"MYD14\": \"C2278858993-LPCLOUD\",\n",
+    "    \"MCD43A3\": \"C2278860820-LPCLOUD\",\n",
+    "    \"MOD13Q1\": \"C1748066515-LPCLOUD\",\n",
+    "    \"MYD13Q1\": \"C2307290656-LPCLOUD\",\n",
+    "    \"MOD13A3\": \"C2327962326-LPCLOUD\",\n",
+    "    \"MYD13A3\": \"C2327957988-LPCLOUD\",\n",
+    "    \"MYD09GQ\": \"C2343109950-LPCLOUD\",\n",
+    "    \"MCD15A3H\": \"C2343110937-LPCLOUD\",\n",
+    "    \"ASTGTM\": \"C1711961296-LPCLOUD\",\n",
+    "    \"ASTGTM_NC\": \"C2439422590-LPCLOUD\",\n",
+    "    \"MOD09GQ\": \"C2343115666-LPCLOUD\",\n",
+    "    \"MOD09A1\": \"C2343111356-LPCLOUD\",\n",
+    "    \"MOD16A2\": \"C2343113232-LPCLOUD\",\n",
+    "    \"MOD09Q1\": \"C2343112831-LPCLOUD\",\n",
+    "    \"MYD09Q1\": \"C2343114343-LPCLOUD\",\n",
+    "    \"MYD09A1\": \"C2343113743-LPCLOUD\",\n",
+    "    \"MOD11_L2\": \"C2343115255-LPCLOUD\",\n",
+    "    \"MYD11_L2\": \"C2343114808-LPCLOUD\",\n",
+    "    \"Daymet_Daily_V4R1_2129\": \"C2532426483-ORNL_CLOUD\",\n",
+    "    \"MODIS_A-JPL-L2P-v2019.0\": \"C1940473819-POCLOUD\",\n",
+    "    \"OSCAR_L4_OC_third-deg\": \"C2036880665-POCLOUD\",\n",
+    "    \"MUR-JPL-L4-GLOB-v4.1\": \"C1996881146-POCLOUD\",\n",
+    "    \"VIIRS_NPP-STAR-L3U-v2.80\": \"C2147485059-POCLOUD\"\n",
     "}\n",
     "\n",
     "provider_list = {\n",
     "    \"ASF\", \"GES_DISC\", \"LAADS\", \"LPCLOUD\", \"POCLOUD\", \"GHRC_DAAC\", \"ORNL_CLOUD\", \"NSIDC_CPRD\"\n",
     "}"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "f23f75d6",
+   "metadata": {},
+   "source": [
+    "## Scoring helper functions:\n",
+    "\n",
+    "The main entry point for the cell below is `score_all_collections`, which will iterate through all rows and calculate both an ideal and an actual score for each collection."
+   ]
+  },
```

Comment:
So the code in the function below got broken down into a lot of functions. (Usually good in service code, but maybe not so good in a notebook). I'm hoping, though, the level of granularity will make individual tweaks to definitions easier (like scores or what constitutes a compatible file format).

Also @asteiker - I know we discussed whether it was worth filtering out legacy formats. I haven't done that in this change, but that would go in `collection_has_compatible_native_format`. (That function considers all listed native formats, for reference)